File: C++/polya.cpp
// import math, time
// from multiprocessing.dummy import Pool as ThreadPool 

#include<iostream>
#include <iomanip>
#include <fstream>
#include <string>
#include <sstream>
#include <math.h> 
#include <algorithm>
#include<vector>
// #include<thread>
// using std::thread;
using std::vector;
using std::sort;
using std::ifstream;
using std::iostream;
using std::to_string;
using std::fixed;
using std::setprecision;
using std::stringstream;
using std::cout; 
using std::endl; 
using std::boolalpha;


long o=0;
long e=0;
void primes(long n){
	long ogn;
	ogn = n;
	if (n%1000000==0 || n==906150256 || n==906150257){
		cout<<n<<endl;
	}
	vector<long> primfac;
	while (n%2 == 0)
    {
        n = n/2;
        primfac.push_back(n);
    }
    for (int i = 3; i <= sqrt(n); i = i+2)
    {
        while (n%i == 0)
        {
            primfac.push_back(i);
            n = n/i;
        }
    }
    if (n > 2)
	    primfac.push_back(n);
	if (primfac.size()%2==0){
		e++;
	}
	else{
		o++;
	}
	if(e>o && n!=1){
		cout<<ogn<<" Winner Winner Chicken Dinner";
		exit(0);
	}
	}

int main(){
	for (int i = 1; i <= 1000000; ++i)
	{
		primes(i);
	}
	if (o>=e){
	cout<<("Polyas: True")<<endl;
	cout<<"Odd: "<< o << endl << "Even: " << e;
	}
}

File: Python/polya.py
import math, time
from multiprocessing.dummy import Pool as ThreadPool 

o=0
e=0
def primes(n):
	global e
	global o
	# print(n)
	if n%100==0:
		print(n)
	# if n == 1:	
	# 	e+=1
	primfac = []
	d = 2
	while d*d <= n:
		while (n % d) == 0:
			primfac.append(d)  # supposing you want multiple factors repeated
			n //= d
		d += 1
	if n > 1:
		primfac.append(n)

		# print(primfac)
	if len(primfac)%2==0:
		e+=1
	else:
		o+=1

# pool = ThreadPool(1000) 
# results = pool.map(primes, range(1,1000))
# results = pool.map(primes, range(1,100000000))
# results = pool.map(primes, range(100000001,200000000))
# results = pool.map(primes, range(200000001,300000000))
# results = pool.map(primes, range(300000001,400000000))
# results = pool.map(primes, range(400000001,500000000))
# results = pool.map(primes, range(500000001,600000000))
# results = pool.map(primes, range(600000001,700000000))
# results = pool.map(primes, range(700000001,800000000))
# results = pool.map(primes, range(800000001,900000000))
# results = pool.map(primes, range(900000001,1000000000))
for s in range(0,1000000):
	primes(s)
if (o>=e):
	print("Polyas: True")
print("Odd: "+str(o)+" Even: "+str(e))
